src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectGCDumpActionTests.cs

```diff
@@ -0,0 +1,127 @@
+                {
+                    IOptionsMonitor<CollectionRuleOptions> ruleOptionsMonitor = host.Services.GetService<IOptionsMonitor<CollectionRuleOptions>>();
+                    CollectGCDumpOptions options = (CollectGCDumpOptions)ruleOptionsMonitor.Get(DefaultRuleName).Actions[0].Settings;
+
+                    ICollectionRuleActionProxy action;
+                    Assert.True(host.Services.GetService<ICollectionRuleActionOperations>().TryCreateAction(KnownCollectionRuleActions.CollectGCDump, out action));
+
+                    EndpointInfoSourceCallback callback = new(_outputHelper);
+                    await using var source = _endpointUtilities.CreateServerSource(out string transportName, callback);
+                    source.Start();
+
+                    AppRunner runner = _endpointUtilities.CreateAppRunner(transportName, TargetFrameworkMoniker.Net60); // Arbitrarily chose Net60
+
+                    Task<IEndpointInfo> newEndpointInfoTask = callback.WaitForNewEndpointInfoAsync(runner, CommonTestTimeouts.StartProcess);
+
+                    await runner.ExecuteAsync(async () =>
+                    {
+                        IEndpointInfo endpointInfo = await newEndpointInfoTask;
+
+                        using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(CommonTestTimeouts.GCDumpTimeout);
+                        CollectionRuleActionResult result = await action.ExecuteAsync(options, endpointInfo, cancellationTokenSource.Token);
+
+                        Assert.NotNull(result.OutputValues);
+                        Assert.True(result.OutputValues.TryGetValue(CollectionRuleActionConstants.EgressPathOutputValueName, out string egressPath));
+                        Assert.True(File.Exists(egressPath));
+
+                        using FileStream gcdumpStream = new(egressPath, FileMode.Open, FileAccess.Read);
+                        Assert.NotNull(gcdumpStream);
+
+                        await ValidateGCDump(gcdumpStream);
+
+                        await runner.SendCommandAsync(TestAppScenarios.AsyncWait.Commands.Continue);
+                    });
+                });
+            }
+            finally
+            {
+                try
+                {
+                    uniqueEgressDirectory?.Delete(recursive: true);
+                }
+                catch
+                {
+                }
+            }
+        }
+
+        private static async Task ValidateGCDump(Stream gcdumpStream)
+        {
+            byte[] buffer = new byte[24];
+
+            const string knownHeaderText = "!FastSerialization.1";
+
+            // Read enough to deserialize known header text.
+            int read;
+            int total = 0;
+            using CancellationTokenSource cancellation = new(CommonTestTimeouts.GCDumpTimeout);
+            while (total < buffer.Length && 0 != (read = await gcdumpStream.ReadAsync(buffer, total, buffer.Length - total, cancellation.Token)))
```

Comment:
Since this is the same as in the dump utility, can you refactor this into an extension method? See https://github.com/dotnet/diagnostics/blob/main/src/Microsoft.Diagnostics.NETCore.Client/StreamExtensions.cs for an example.